### PR TITLE
Domain events refactoring

### PR DESCRIPTION
**Pull request by rprzedzik**

### CODE REVIEW

File: module/category-tree/src/Domain/Event/CategoryTreeCreatedEvent.php [hunk 1]
```diff
@@ -52,13 +53,15 @@ public function __construct(CategoryTreeId $id, string $code, TranslatableString
...
-     * @return CategoryTreeId
+     * @return CategoryTreeId|AbstractId
      */
-    public function getId(): CategoryTreeId
+    public function getAggregateId(): AbstractId
     {
         return $this->id;
     }
 
+
```
Comment (wiewiurdp) [inline module/category-tree/src/Domain/Event/CategoryTreeCreatedEvent.php]: are these lines needed ?

File: module/account/src/Domain/Event/Role/RolePrivilegesChangedEvent.php [hunk 2]
```diff
@@ -33,18 +42,28 @@ class RolePrivilegesChangedEvent implements DomainEventInterface
     private $to;
 
     /**
+     * @param RoleId      $id
      * @param Privilege[] $from
      * @param Privilege[] $to
      */
-    public function __construct(array $from, array $to)
+    public function __construct(RoleId $id, array $from, array $to)
```
Comment (BastekBielawski) [inline module/account/src/Domain/Event/Role/RolePrivilegesChangedEvent.php]: On construct you require `RoleId` and in `getAggregateId` u allow to return `AbstractId`. Why u don't set `AbstractId` in constructor?
PHP 7.4 allow to set in interface AbstractId as return type and in base class u can set RoleId as return type. PHP will compile it without error :)
When u setup constructor as `__construct(AbstractId $id, array $from, array $to)` u don't need to setup constructor that many times :)

File: module/account/src/Resources/config/services.yml [hunk 3]
```diff
@@ -34,6 +34,10 @@ services:
         resource: '../../Infrastructure/Handler/*'
         tags: ['messenger.message_handler']
 
+    Ergonode\Account\Persistence\Dbal\Projector\:
```
Comment (BastekBielawski) [inline module/account/src/Resources/config/services.yml]: I love it! Finally! :D

File: module/attribute/src/Persistence/Dbal/Projector/Attribute/AttributeCreatedEventProjector.php [hunk 4]
```diff
@@ -10,17 +10,14 @@
...
-use Ergonode\EventSourcing\Infrastructure\DomainEventInterface;
-use Ergonode\EventSourcing\Infrastructure\Exception\UnsupportedEventException;
-use Ergonode\EventSourcing\Infrastructure\Projector\DomainEventProjectorInterface;
 use JMS\Serializer\SerializerInterface;
 use Ramsey\Uuid\Uuid;
 
 /**
  */
-class AttributeCreatedEventProjector implements DomainEventProjectorInterface
+class AttributeCreatedEventProjector
```
Comment (BastekBielawski) [inline module/attribute/src/Persistence/Dbal/Projector/Attribute/AttributeCreatedEventProjector.php]: Separate it to different methods :D Invoke method look's like old fashion portal class :D

File: module/attribute/src/Persistence/Dbal/Projector/Attribute/AttributeHintChangedEventProjector.php [hunk 5]
```diff
@@ -10,16 +10,14 @@
...
 use Ergonode\Core\Domain\Entity\AbstractId;
-use Ergonode\EventSourcing\Infrastructure\DomainEventInterface;
-use Ergonode\EventSourcing\Infrastructure\Exception\UnsupportedEventException;
-use Ergonode\EventSourcing\Infrastructure\Projector\DomainEventProjectorInterface;
 use Ramsey\Uuid\Uuid;
 
 /**
  */
-class AttributeHintChangedEventProjector implements DomainEventProjectorInterface
+class AttributeHintChangedEventProjector
```
Comment (BastekBielawski) [inline module/attribute/src/Persistence/Dbal/Projector/Attribute/AttributeHintChangedEventProjector.php]: Try to separate it to different methods. Invoke is huge :)

File: module/workflow/src/Persistence/Dbal/Repository/DbalWorkflowRepository.php [hunk 6]
```diff
@@ -27,18 +27,20 @@ class DbalWorkflowRepository implements WorkflowRepositoryInterface
...
-     * @var DomainEventDispatcherInterface
+     * @var EventBusInterface
      */
-    private $eventDispatcher;
+    private $eventBus;
 
     /**
-     * @param DomainEventStoreInterface      $eventStore
-     * @param DomainEventDispatcherInterface $eventDispatcher
+     * DbalWorkflowRepository constructor.
```
Comment (BastekBielawski) [inline module/workflow/src/Persistence/Dbal/Repository/DbalWorkflowRepository.php]: `DbalWorkflowRepository constructor.` this is unnecessary

File: module/workflow/src/Resources/config/services.yml [hunk 7]
```diff
@@ -25,7 +25,10 @@ services:
 
     Ergonode\Workflow\Infrastructure\Handler\:
         resource: '../../Infrastructure/Handler/*'
-        exclude: '../../Infrastructure/Handler/{Strategy}'
+        tags: ['messenger.message_handler']
+
+    Ergonode\Workflow\Persistence\Dbal\Projector\:
```
Comment (BastekBielawski) [inline module/workflow/src/Resources/config/services.yml]: I think this can be done globaly

**Pull request merged**